### ISSUE TITLE
Undefined property: WP_Post_Type::$ID in single cpt

### ISSUE THREAD

**Issue by suetin** [Log1x/crumb#17]
Undefined property: WP_Post_Type::$ID
vendor\log1x\crumb\src\Crumb.php(line 231)
Change $type->ID to get_queried_object_id()
Its work for me

**Comment by Log1x** [comment]
can you dump `$type` and see if the ID is on it at all? i dont have a test env at the moment.

**Comment by suetin** [comment]
`$type = get_post_type_object(
            get_post_type()
        );

        if (! empty($type)) {
            dd($type);
            $this->add(
                $type->label,
                get_post_type_archive_link($type->name),
                $type->ID
            );
        }`

` WP_Post_Type {#10175 ▼
  +name: "product"
  +label: "Каталог"
  +labels: {#10180 ▶}
  +description: ""
  +public: true
  +hierarchical: false
  +exclude_from_search: false
  +publicly_queryable: true
  +show_ui: true
  +show_in_menu: true
  +show_in_nav_menus: true
  +show_in_admin_bar: true
  +menu_position: 6
  +menu_icon: "dashicons-database"
  +capability_type: "page"
  +map_meta_cap: true
  +register_meta_box_cb: null
  +taxonomies: []
  +has_archive: true
  +query_var: "product"
  +can_export: true
  +delete_with_user: null
  +template: []
  +template_lock: false
  +_builtin: false
  +_edit_link: "post.php?post=%d"
  +cap: {#10177 ▶}
  +rewrite: array:5 [▶]
  +show_in_rest: true
  +rest_base: false
  +rest_namespace: "wp/v2"
  +rest_controller_class: false
  +rest_controller: null
  +"site_filters": null
  +"site_sortables": null
  +"show_in_feed": false
  +"archive": null
  +"featured_image": null
  +"quick_edit": true
  +"admin_cols": array:2 [▶]
  +"admin_filters": array:1 [▶]
}`

**Comment by suetin** [comment]
![image](https://github.com/Log1x/crumb/assets/54860941/d8bd4743-d281-4f7c-b612-8094d797b477)

This code works correctly

**Comment by Log1x** [comment]
Yeah that seem's right. I'm a little confused how I got to the current implementation. 😊 

Care to do a PR?

**Comment by suetin** [comment]
Yes🙂
